project/project.go

```diff
@@ -16,8 +16,72 @@ import (
 	"github.com/machinefi/sprout/vm"
 )
 
+var (
+	errEmptyConfigData    = errors.New("empty config data")
+	errInvalidProjectCode = errors.New("invalid project code")
+	errUnsupportedVMType  = errors.New("unsupported vm type")
+)
+
 type Config struct {
-	DatasourceURI   string            // TODO change this after contract code updated
+	Datasource     string        `json:"datasource"`
+	DefaultVersion string        `json:"defaultVersion"`
+	Versions       []*ConfigData `json:"versions"`
+	defaultData    *ConfigData
+	versions       map[string]*ConfigData
+}
+
+func (c *Config) Validate() error {
+	if len(c.Versions) == 0 {
+		return errEmptyConfigData
+	}
+
+	// remove invalid version data and set default
+	var firstVersionData *ConfigData
+	for i, v := range c.Versions {
+		if err := v.Validate(); err != nil {
+			c.Versions = append(c.Versions[:i], c.Versions[i+1:]...)
+			continue
+		}
+		if c.versions == nil {
+			c.versions = make(map[string]*ConfigData)
+		}
+		if _, ok := c.versions[v.Version]; ok {
+			continue // to avoid overwrite exist version data
+		}
+		c.versions[v.Version] = v
+		if firstVersionData == nil {
+			firstVersionData = v
+		}
+	}
+
+	// ensure contains at least one valid version data
+	if firstVersionData == nil {
+		return errEmptyConfigData
+	}
+
+	c.defaultData = c.versions[c.DefaultVersion]
+	if c.defaultData == nil {
+		c.defaultData = firstVersionData
+	}
+
+	// inherit default values
+	if c.defaultData.DatasourceURI == "" {
+		c.defaultData.DatasourceURI = c.Datasource
+	}
+
+	return nil
+}
+
+func (c *Config) DefaultConfigData() *ConfigData {
+	return c.defaultData
+}
+
+func (c *Config) GetConfigDataByVersion(version string) *ConfigData {
+	return c.versions[version]
+}
+
+type ConfigData struct {
+	DatasourceURI   string            `json:"datasourceURI"`
```

Comment:
DatasourceURI should in Config field, for a project should only have one data source at a time

Reply:
`ConfigData` will inherit from `Config` if this field is not assigned.